Add a toggle to skip email validation

meain opened the pull request:
This PR adds an option `UseProvidedAddress` which when set directly uses the email provided by the user without validating or trying to parse out its components.

Closes https://github.com/xhit/go-simple-mail/issues/95

meain commented:
@xhit Quick question. I saw that you were using `address.Address` [at certain places](https://github.com/alcionai/go-simple-mail/blob/abcd63ac2d0e87541de479cf261fe295146c7950/email.go#L369) and `address.String()` at [other](https://github.com/alcionai/go-simple-mail/blob/abcd63ac2d0e87541de479cf261fe295146c7950/email.go#L395). Is there a reason why we are doing it differently at different places.

If we use `address.String()` at all the paces, we could also close https://github.com/xhit/go-simple-mail/issues/11 .

xhit commented:
@meain `address.Address` contains only the email, with this we add to recipients to send a SMTP `RCPT` command because we only need the address here. The `address.String()` is the full address, with the name if provided, and the address enclosed by `<` and `>`, used to create a RFC822 formatted message to use un SMTP `DATA` command, also, dev can get it using `GetMessage` method.

Not all addresses in the message are populated by default, the BCC for example, are added to recipients for the `RCPT command, but not in the message for the `DATA` command for security, with this, the recipients will not know the BCC addresses.

xhit commented on email.go:
```diff
@@ -330,16 +338,20 @@ func (email *Email) AddAddresses(header string, addresses ...string) *Email {
-			if err != nil {
-				email.Error = errors.New("Mail Error: " + err.Error() + "; Header: [" + header + "] Address: [" + addresses[i] + "]")
-				return email
+			if !email.UseProvidedAddress {
```
When `email.UseProvidedAddress`, `fullAddress` should be enclosed with `<` and `>`

meain replied:
I wanted to pass in address like `First Last <firstlast@provider.com>` with `email.UseProvidedAddress`. Surrounding this with `<` and `>` would end up with incorrect addresses.

PS: I'm assuming it is valid to have a full address (address with name as well) in all fields according to spec.

xhit replied:
with `email.UseProvidedAddress` we have an ambiguous situation: while `First Last <firstlast@provider.com>` is supported without `email.UseProvidedAddress`, you can have emails without `@`, so for these a check should exists to verify if address is enclosed, this issue exists only because you can have something like `First Last <sysadmin>` according to #95.

About my comment:

> When email.UseProvidedAddress, fullAddress should be enclosed with < and >

I missed you can send the name, my fault.

We can leave this without verification, but we need to warn in the README about this toggle will not complain with RFC 5322, so only the  address is allowed to avoid error sending the email, so, in your case, not `First Last <sysadmin>`, only `sysadmin`, because `RCPT` only needs the address.

About this:

> PS: I'm assuming it is valid to have a full address (address with name as well) in all fields according to spec.

Yes, you can, but in `DATA` for the headers, the `rcpt` method in SMTP receives only the address., this is why I extract the address from a fullAddress with golang mail std package.

In your case, if you send `First Last <sysadmin> with the toggle in `true`, the RCPT will not work, you can't send the email.

meain replied:
Let me make sure I understand your ask.

You are suggesting that we can leave the functionality as currently in the PR, but warn in README that when `email.UseProvidedAddress` is set, the user is responsible to ensure they only send the email address or else `RCPT` will fail.

xhit replied:
Yes. With this PR your use case is unique, so, before merge it, I need to know if the implementation works for you and what were the formats of emails used. Also, if RCPT TO was called and the result (if mail was sent).

I don't know if you need to call the RCPT, but you only need to use this toggle for emails you really know aren't RFC 5322, because if you have `First Last <firstlast@provider.com>` with this toggle, the `RCPT TO` will throw an error too.

meain replied:
> I don't know if you need to call the RCPT

We won't use this package to send emails. The use case for us is purely to generate `.eml` files and write to disk which then gets processes by a different tool.

> ... what were the formats of emails used.

We end up using both formats, ie name + email or email. We need this toggle as we know some of our addresses do no conform to RFC5322. The main problem that we have is that we certain addresses which only contain a name field. The data is incomplete and as I mentioned in the original issue, happens primarily in draft emails that we ended up saving. In these cases, we want the address fields in the final `.eml` file to just contain a name. I understand these emails cannot be sent, but our usecase is a bit different.

meain commented:
> `address.Address` contains only the email, with this we add to recipients to send a `SMTP` `RCPT` command because we only need the address here.

Hmm, I see. Would you be open to changing it to accepting the full address. Would it cause issues?

xhit commented:
> Hmm, I see. Would you be open to changing it to accepting the full address. Would it cause issues?

I can't, RCPT only receives the address, not full address.

meain commented:
@xhit Following our discussion, I've added a note in the readme and rebased the code on latest master. Let me know if you need anything else from the PR.

xhit commented:
@meain sorry, I'm currently on vacations, I will tag a new version after that, you can use master branch. Thanks!

meain commented:
Thanks @xhit. No rush on the tag :D
